### DISCUSSION TITLE
Parent branch

### DISCUSSION THREAD

**Discussion by ahmedfourti** [renovatebot/renovate#15946] [Request Help]
Hello guys,

In my repo I have multiple branches.
I configured renovate to create MR on branch "dev".
But I am wondering from witch branch is it creating that new branch ? is it possible to configure it ?
I need renovate to create new branch from "main" and create MR on branch "dev"

Thank you

**Comment by rarkins** [comment]
No, that's not possible. There's nothing guaranteeing that two branches like that even have a shared history.

**Reply by ahmedfourti** [reply]
But which branch does Renovate use to create the MR ?

**Reply by rarkins** [reply]
Renovate creates a single commit off the baseBranch (which is default branch unless configured otherwise) and then creates a MR targetting that same base branch. i.e. the "diff" should always be a single commit

**Reply by ahmedfourti** [reply]
But in my case the default branch is "main" and I configured baseBranch to "dev".
So if I understood well, Renovate will create a MR targetting dev branch but the source is "main" ?

**Reply by rarkins** [reply]
No.

If default branch is `main` then the config will always be ready from it (e.g. `renovate.json`) but if you configure `"baseBranches": ["dev"]` then Renovate will create all its branches off `dev` and target `dev` with MRs.

**Reply by ahmedfourti** [reply]
All clear, thanks for your help :)
